march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -316,4 +339,18 @@ void MarchHardwareInterface::updatePowerNet()
     }
   }
 }
+
+void MarchHardwareInterface::updateIMotionCubeState(int jointIndex)
```

Comment:
I suggest wrapping this in another method that updates it for all joints for cality